views/js/pciCreator/dev/textReaderInteraction/runtime/textReaderInteraction.amd.js

```diff
@@ -0,0 +1,168 @@
+                    pci.widgetRenderer.renderChoices(that._taoCustomInteraction.properties);
+                });
+
+                this.$container.on('click', '.js-remove-choice', function () {
+                    var num = $(this).data('choice-index');
+                    properties.choices.splice(num, 1);                    
+                    pci.widgetRenderer.renderChoices(that._taoCustomInteraction.properties);
+                });
+
+                //add page event
+                this.$container.on('click', '[class*="js-add-page"]', function () {
+                    var num = properties.pages.length + 1,
+                        $button = $(this),
+                        pageData = {
+                            label : 'Page ' + num,
+                            content : 'page ' + num + ' content',
+                            id : ++maxPageId
+                        };
+
+                    if ($button.hasClass('js-add-page-before')) {
+                        properties.pages.unshift(pageData);
+                    } else if ($button.hasClass('js-add-page-after')) {
+                        properties.pages.push(pageData);
+                    }
+                    pci.widgetRenderer.renderPages(properties);
+                    that.trigger('pageschange');
+                });
+
+                //remove page event
+                this.$container.on('click', '.js-remove-page', function () {
+                    var $button = $(this),
+                        tabNum = $button.data('page-num');
+                    properties.pages.splice(tabNum, 1);
+                    pci.widgetRenderer.renderPages(properties);
+                    that.trigger('pageschange');
+                });
+
+                //page navigation events
+                this.$container.on('click', '.js-prev-page, .js-next-page', function () {
+                    var $button = $(this),
+                        direction = $button.hasClass('js-next-page') ? 1 : -1,
+                        currentPage = pci.widgetRenderer.tabsManager.index(),
+                        index = currentPage + direction;
+
+                    if (index >= 0 && properties.pages.length > index) {
+                        pci.widgetRenderer.tabsManager.index(index);
+                    }
+                });
+            },
+            /**
+             * Programmatically set the response following the json schema described in
+             * http://www.imsglobal.org/assessment/pciv1p0cf/imsPCIv1p0cf.html#_Toc353965343
+             * 
+             * @param {Object} interaction
+             * @param {Object} response
+             */
+            setResponse : function (response) {
+                var value = response && response.base ? parseInt(response.base.integer, 10) : [];
+                $.each(value, function (key, val) {
+                    this.$container.find('input[value="' + val + '"]').prop('checked', true);
+                });
+            },
```

Comment:
you may wonder what the setResponse() will be doing in this context ... well nothing actually ! you can leave this function empty for now and add a comment stating the reason why.